### PR TITLE
Emphasize that `Welcome` is not a constructor

### PR DESCRIPTION
**Pull request by stevemao**
**Before submitting a pull request,** please make sure the following is done:

1. Fork [the repository](https://github.com/facebook/react) and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).
5. Make sure your code lints (`npm run lint`).
6. Run the [Flow](https://flowtype.org/) typechecks (`npm run flow`).
7. If you added or removed any tests, run `./scripts/fiber/record-tests` before submitting the pull request, and commit the resulting changes.
8. If you haven't already, complete the [CLA](https://code.facebook.com/cla).

### CODE REVIEW

**Comment by stevemao** [comment]
See https://twitter.com/dan_abramov/status/838537120780255232, I'm pretty sure there are people like me who got confused.

File: docs/docs/components-and-props.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ function Welcome(props) {
 
 This function is a valid React component because it accepts a single "props" object argument with data and returns a React element. We call such components "functional" because they are literally JavaScript functions.
 
+Note that the capitalization of `Welcome` does not indicate that it is a constructor. [It only indicates that the JSX tag is referring to a React component](https://facebook.github.io/react/docs/jsx-in-depth.html#html-tags-vs.-react-components).
```
Comment (gaearon) [inline docs/docs/components-and-props.md]: I can see this as potentially confusing since we haven't really shown it'll be used in a JSX tag yet.

Comment (stevemao) [inline docs/docs/components-and-props.md]: Maybe reword this to `It is just a way to represent user-defined components`

Comment (gaearon) [inline docs/docs/components-and-props.md]: I'm also not sure that people reading this necessarily know that constructors are often capitalized.

Comment (stevemao) [inline docs/docs/components-and-props.md]: We can link `constructor` to the mdn page. I think people who read this should have some basic knowledges.

Comment (gaearon) [inline docs/docs/components-and-props.md]: I mean that the whole convention of capitalizing constructors has no basis in the language. It's just a convention popularized by JSLint (or JSHint, I don't remember), I think. So for people not exposed to this convention (many React beginners) this implication will only further confuse them.

Comment (stevemao) [inline docs/docs/components-and-props.md]: I'd rather write more to explain it :) It's a really common pattern that's not just specific to javascript. I've always thought that components should be a class/constructor for a long time :)

**Comment by gaearon** [comment]
I still think it’s more confusing to add this than to leave it as is, sorry 😞 . Gotta make the call here, and this is how I see it.
